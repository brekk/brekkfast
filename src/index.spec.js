import Mersy from "./index"

const SEEDS = {
  ONE: "one",
  TWO: 2222,
}
const alphabet = "abcdefghijklmnopqrstuvwxyz".split("")
const alphaObj = []
  .concat(alphabet)
  .map((c, i) => [c, i])
  .reduce((agg, [c, i]) => Object.assign({}, agg, { [c]: i }))

function testWithSeed(seed) {
  describe("Mersy", () => {
    let M1
    let M2
    beforeAll(() => {
      M1 = new Mersy(seed)
      M2 = new Mersy(seed)
    })
    test("integer", () => {
      const range = { min: -500, max: 500 }
      const x = M1.integer(range)
      const y = M2.integer(range)
      expect(x).toEqual(y)
      expect(x).toMatchSnapshot()
      const z = M1.integer(range)
      expect(x).not.toEqual(z)
    })
    test("pick", () => {
      const x = M1.pick(alphabet)
      const y = M2.pick(alphabet)
      expect(x).not.toEqual(y)
      expect(x).toMatchSnapshot()
      expect(y).toMatchSnapshot()
    })
    test("pickKey", () => {
      const x = M1.pickKey(alphaObj)
      const y = M2.pickKey(alphaObj)
      expect(x).not.toEqual(y)
      expect(x).toMatchSnapshot()
      expect(y).toMatchSnapshot()
    })
    test("pickValue", () => {
      const x = M1.pickValue(alphaObj)
      const y = M2.pickValue(alphaObj)
      expect(x).not.toEqual(y)
      expect(x).toMatchSnapshot()
      expect(y).toMatchSnapshot()
    })
  })
}

testWithSeed(SEEDS.ONE)
testWithSeed(SEEDS.TWO)
