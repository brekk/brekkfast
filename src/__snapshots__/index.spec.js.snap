// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mersy integer 1`] = `167`;

exports[`Mersy integer 2`] = `49`;

exports[`Mersy pick 1`] = `"o"`;

exports[`Mersy pick 2`] = `"p"`;

exports[`Mersy pick 3`] = `"t"`;

exports[`Mersy pick 4`] = `"q"`;

exports[`Mersy pickKey 1`] = `"l"`;

exports[`Mersy pickKey 2`] = `"n"`;

exports[`Mersy pickKey 3`] = `"w"`;

exports[`Mersy pickKey 4`] = `"t"`;

exports[`Mersy pickValue 1`] = `0`;

exports[`Mersy pickValue 2`] = `11`;

exports[`Mersy pickValue 3`] = `15`;

exports[`Mersy pickValue 4`] = `22`;
